### ISSUE TITLE
Rule Request: Condition list instead of boolean &&

### ISSUE THREAD

**Issue by SimplyDanny** [realm/SwiftLint#6039]
Swift has the notion of condition lists that behave exactly as if the condition elements were combined with the `&&` operator. Condition lists can be regarded as **idiomatic** Swift.

This new rule should trigger on boolean expressions used in places where condition lists would be allowed, too. It can even auto-transform these expressions. It is very important, though, that the order of evaluation is preserved to avoid changing the code's semantic with respect to side effects.

The rule should be **opt-in**. In fact, a reverse rule triggering on condition lists is also conceivable. At the moment, there is **no specific configuration** necessary.

#### Triggering examples

```swift
if a && b {}
while (a || b) && c {}
```

can be replaced with

```swift
if a, b {}
while a || b, c {}
```

#### Non-triggering examples

```swift
do {} while a || b && c
if a, { b && c }}() {}
```

**Comment by suojae** [comment]
Hi @SimplyDanny! My colleague @hyeffie and I are interested in implementing this rule. Would it be okay if we take on this issue? We plan to complete the PR by the end of this week.

**Comment by SimplyDanny** [comment]
> Hi @SimplyDanny! My colleague @hyeffie and I are interested in implementing this rule. We're currently working on the additional UIColor magic numbers PR (#6035), but would also like to take on this issue in parallel. Would it be okay if we implement this as well? We plan to complete the PR by the end of this week

Happy to hear that! Please take your time. There is no time pressure here. 😉

I'd like to point you to the documentation for contributors, especially [the section about adding new rules](https://github.com/realm/SwiftLint/blob/main/CONTRIBUTING.md#Rules). I'm very keen to hear how the pretty new `swiftlint-dev` command worked for you. 

Please let me know and have fun implementing the new rule!

**Comment by suojae** [comment]
Thank you! I'll definitely check out the documentation, especially the section about adding new rules. We haven't used the `swiftlint-dev` command yet, but I'm looking forward to trying it out and will share our experience with it.

We'll keep you updated on our progress. Thanks again for the support! 😆
